.github/workflows/deploy_docker_image.yml

```diff
@@ -0,0 +1,86 @@
+name: Deploy Harmony GDAL Adapter Docker image
+
+on:
+  push:
+    branches: [ main ]
```

Comment:
I split the triggers that were in the test workflow so that the test-only workflow runs only for PRs against `main`. Then this new workflow (which tests, then deploys) runs only for changes to `main`.